packages/blocks/src/lib/maplibre/marker.svelte

```diff
@@ -26,15 +26,22 @@ export let scale = 1;
 /** The marker color. */
 export let color = '';
 
+/** The marker icon */
+export let icon: HTMLElement | undefined;
+
+/** The marker rotation, where 0 is north */
+export let rotation = 0;
+
 const { map } = useMapLibre();
 
 let marker: Marker | undefined;
 
 $: {
   marker?.remove();
-  marker = new Marker({ scale, color });
+  marker = new Marker({ element: icon!, scale, color });
```

Comment:
To avoid a non null assertion I'd recommend:
```suggestion
  marker = new Marker(element ? { element, scale, color } : { scale, color });